frontend/app/utils/env.server.ts

```diff
@@ -0,0 +1,47 @@
+import { z } from 'zod';
+
+/**
+ * Environment variables that will be available to server only.
+ */
+const serverEnv = z.object({
+  NODE_ENV: z.enum(['production', 'development', 'test']),
+  LOG_LEVEL: z.enum(['error', 'warn', 'info', 'http', 'verbose', 'debug', 'silly']).default('info'),
+  LANG_COOKIE_NAME: z.string().default('_gc_lang'),
+  LANG_COOKIE_DOMAIN: z.string().optional(),
+  LANG_COOKIE_PATH: z.string().default('/'),
+  LANG_COOKIE_HTTP_ONLY: z.coerce.boolean().default(true),
+  LANG_COOKIE_SECURE: z.coerce.boolean().default(true),
+  LANG_QUERY_PARAM: z.string().default('lang'),
+});
+
+export type ServerEnvKey = keyof z.infer<typeof serverEnv>;
+
+/**
+ * Environment variables that will be available to client.
+ * ⚠️ IMPORTANT: DO NOT PUT SENSITIVE CONFIGURATIONS HERE ⚠️
+ */
+const clientEnv = serverEnv.pick({
+  LANG_QUERY_PARAM: true,
+});
+
+export type ClientEnvKey = keyof z.infer<typeof clientEnv>;
+
+/**
+ * Return the server property value associated with the given key.
+ */
+export function getEnv<T>(key: ServerEnvKey) {
+  const result = serverEnv.safeParse(process.env);
+
+  if (!result.success) {
+    throw new Error(`Invalid application configuration: ${result.error}`);
+  }
+
+  return result.data[key] as T;
+}
```

Comment:
Return the parsed config and let the consumer use the config key with destructuring assignment if he need a specific key... that way everything will be typed.

```suggestion
export function getEnv() {
  const result = serverEnv.safeParse(process.env);

  if (!result.success) {
    throw new Error(`Invalid application configuration: ${result.error}`);
  }

  return result;
}
```

Reply:
When I remove the `<T>` generic, this happens...

This should be a `string`:

![image](https://github.com/DTS-STN/canadian-dental-care-plan/assets/48123208/33964849-c921-4cf7-810a-4b753a8b8c81)

These values' types should be tighter:

![image](https://github.com/DTS-STN/canadian-dental-care-plan/assets/48123208/a8b237ca-6d87-4f74-8cde-0c1737bfd68f)

Reply:
I fixed this as per your suggestions. Please re-review.